bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyEventServlet.java

```diff
@@ -0,0 +1,163 @@
+            logger.debug("Shelly: CallbackServlet started at '{}'", SHELLY_CALLBACK_URI);
+        } catch (RuntimeException | NamespaceException | ServletException e) {
+            logger.warn("Shelly: Could not start CallbackServlet: {} ({})", e.getMessage(), e.getClass());
+        }
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        if (httpService != null) {
+            httpService.unregister(SHELLY_CALLBACK_URI);
+        }
+        logger.debug("Shelly: CallbackServlet stopped");
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    protected void service(@Nullable HttpServletRequest request, @Nullable HttpServletResponse resp)
+            throws ServletException, IOException {
+        String data = inputStreamToString(request);
+        String path = request.getRequestURI().toLowerCase();
+        String deviceName = "";
+        String index = "";
+        String type = "";
+
+        try {
+            String ipAddress = request.getHeader("HTTP_X_FORWARDED_FOR");
+            if (ipAddress == null) {
+                ipAddress = request.getRemoteAddr();
+            }
+            Map<String, String[]> parameters = request.getParameterMap();
+            logger.debug("CallbackServlet: {} Request from {}:{}{}?{}", request.getProtocol(), ipAddress,
+                    request.getRemotePort(), path, parameters.toString());
+            if (!path.toLowerCase().startsWith(SHELLY_CALLBACK_URI) || !path.contains("/event/shelly")) {
+                logger.info("ERROR: CallbackServletreceived unknown request- path = {}, data={}", path, data);
+                return;
+            }
+
+            // URL looks like
+            // <ip address>:<remote port>/shelly/event/shellyrelay-XXXXXX/relay/n?xxxxx or
+            // <ip address>:<remote port>/shelly/event/shellyrelay-XXXXXX/roller/n?xxxxx or
+            // <ip address>:<remote port>/shelly/event/shellyht-XXXXXX/sensordata?hum=53,temp=26.50
+            deviceName = StringUtils.substringBetween(path, "/event/", "/").toLowerCase();
+            if (path.contains("/" + EVENT_TYPE_RELAY + "/") || path.contains("/" + EVENT_TYPE_ROLLER + "/")
+                    || path.contains("/" + EVENT_TYPE_LIGHT + "/")) {
+                index = StringUtils.substringAfterLast(path, "/").toLowerCase();
+                type = StringUtils.substringBetween(path, deviceName + "/", "/" + index);
+            } else {
+                index = "";
+                type = StringUtils.substringAfterLast(path, "/").toLowerCase();
+            }
+            logger.trace("Process event of type type={} for device {}, index={}", type, deviceName, index);
+            Map<String, String> parms = new HashMap<String, String>();
+            for (Map.Entry<String, String[]> p : parameters.entrySet()) {
+                parms.put(p.getKey(), p.getValue()[0]);
+
+            }
+            handlerFactory.onEvent(deviceName, index, type, parms);
+
+        } catch (RuntimeException e) {
```

Comment:
do not catch runtime exceptions

Reply:
see comment on RuntimeExceptions as part of today's build